internal/service/dns_config/api_delegation_resource_test.go

```diff
@@ -0,0 +1,446 @@
+package dns_config_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/dns_config"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+func TestAccDelegationResource_basic(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationBasicConfig("test.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test.123."),
+					resource.TestCheckResourceAttr(resourceName, "delegation_servers.#", "1"),
+					// Test Read Only fields
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "protocol_fqdn"),
+					resource.TestCheckResourceAttrSet(resourceName, "parent"),
+					// Test fields with default value
+					resource.TestCheckResourceAttr(resourceName, "disabled", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDelegationResource_disappears(t *testing.T) {
+	resourceName := "bloxone_dns_delegation.test"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckDelegationDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDelegationBasicConfig("test.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					testAccCheckDelegationDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccDelegationResource_Comment(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test_comment"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationComment("test.123.", "COMMENT_REPLACE_ME", "123."),
```

Comment:
can you pls change COMMENT_REPLACE_ME :)

Reply:
Haha, Yes I was working on the UTs. Will push the change with updated comment